build.gradle

```diff
@@ -0,0 +1,265 @@
+buildscript {
+    repositories {
+        mavenLocal()
+        mavenCentral()
+        jcenter()
+        maven { url "http://repository.jboss.org/nexus/content/groups/public/" }
+        maven { url 'https://repo.spring.io/plugins-release' }
+        maven { url "http://maven.research.rackspacecloud.com/content/repositories/releases" }
+        maven { url "https://plugins.gradle.org/m2/" }
+        maven { url "http://repository.jboss.org/nexus/content/groups/public/" }
+    }
+    dependencies {
+        classpath 'org.hibernate.build.gradle:gradle-maven-publish-auth:2.0.1'
+        classpath 'org.springframework.build.gradle:propdeps-plugin:0.0.7'
+        classpath "com.github.jacobono:gradle-jaxb-plugin:1.3.5"
+        classpath "org.unbroken-dome.gradle-plugins:gradle-testsets-plugin:1.0.2"
+
+    }
+
+}
+
+configure(allprojects) {
+    apply plugin: 'propdeps'
+    apply plugin: 'propdeps-maven'
+    apply plugin: 'propdeps-idea'
+    apply plugin: 'scala'
+}
+
+configurations.all {
+    transitive = false
+}
+
+allprojects {
+    apply plugin: 'idea'
+
+    tasks.withType(ScalaCompile) {
+        scalaCompileOptions.useAnt = false
+    }
+    sourceCompatibility = '1.7'
+    targetCompatibility = '1.7'
+}
+
+subprojects {
+    apply plugin: 'scala'
+    group = 'org.openrepose.filters.impersonation'
+    version = '3.0'
+
+    repositories {
+        mavenLocal()
+        mavenCentral()
+        maven { url "http://maven.research.rackspacecloud.com/content/repositories/releases" }
+        maven { url "http://repo.typesafe.com/typesafe/maven-releases" }
+        maven { url "https://maven.research.rackspacecloud.com/content/groups/aggregate" }
+    }
+
+    dependencies {
+    }
+
+    tasks.withType(ScalaCompile) {
+        scalaCompileOptions.useAnt = false
+    }
+
+    task 'create-dirs' << {
+        sourceSets.all { set ->
+            set.allSource.srcDirs.each { it.mkdirs() }
+        }
+    }
+}
+
+
+/**
+ * Because an Ear project is always the same, this is a function to clean up our build script
+ * @param projectName The name of the ear project
+ * @param deps a list of dependencies that will go in this ear
+ * @return
+ */
+def filterBundleProject(String projectName, List<String> deps) {
+    project(projectName) {
+        apply plugin: 'ear'
+        apply plugin: 'maven'
+        apply plugin: 'maven-publish'
+        apply plugin: 'maven-publish-auth'
+
+        configurations {
+            earlib
+        }
+
+        dependencies {
```

Comment:
This stuff has to do with how to get all the dependencies into an ear file properly. I had to do it in the past on gradle stuff for repose when I trialed a few things out, and it's how we did it in the testing filter bundles.

You depend on the project's dependencies that this ear buliding project depends on in the right way. It's kind of annoying, but without it you don't get the transitive dependencies of your project packaged in the ear's libs dir.